src/main/java/com/ibanity/signature_service/services/SignatureService.java

```diff
@@ -0,0 +1,43 @@
+package com.ibanity.signature_service.services;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.MGF1ParameterSpec;
+import java.security.spec.PSSParameterSpec;
+import java.util.Base64;
+
+import com.ibanity.signature_service.services.exception.EncryptionException;
+
+public class SignatureService {
+
+    private static final String SIGNATURE_ALGORITHM = "RSASSA-PSS";
+    private static final PSSParameterSpec PARAMETER_SPEC = new PSSParameterSpec("SHA-256", "MGF1",
+            MGF1ParameterSpec.SHA256, 32, 1);
+
+    private final PrivateKey privateKey;
+
+    public SignatureService(PrivateKey privateKey) {
+        this.privateKey = privateKey;
+    }
+
+    public String sign(String toSign) {
```

Comment:
This is globally an HTTP Signature. There are other method to sign HTTP request, such as JWS. In each type of signature (http-signature or jws) there are different algorithm to support. Glad to share my point of view about this

Reply:
I mean that there are multiple ways of signing request: HttpSignature is one. JWS is another one.
In which case I would have setup to lib structure so that http-signature is in scope, (even if we don't have or support JWS at this stage)